server/src/test/java/org/elasticsearch/search/query/QueryPhaseTests.java

```diff
@@ -703,6 +703,23 @@ public void testNumericLongOrDateSortOptimization() throws Exception {
         searchContext.sort(sortAndFormats);
         QueryPhase.executeInternal(searchContext);
         assertSortResults(searchContext.queryResult().topDocs().topDocs, (long) numDocs, true);
+
+        // 5. Test that sort optimization is NOT run with size 0
+        {
+            sortAndFormats = new SortAndFormats(longSort, new DocValueFormat[]{DocValueFormat.RAW});
+            searchContext = spy(new TestSearchContext(null, indexShard, newContextSearcher(reader)));
```

Comment:
I'm wondering why we use a `spy` here, if we don't verify the calls after?

Reply:
@jtibshirani Thanks for the review.  I am using `spy` here to wrap an instance of `searchContext` for it to return a fake `mapperService` on the following line of 710.

Reply:
Got it, thanks.